tests/ui_tests/test_ui_fields.py

```diff
@@ -0,0 +1,73 @@
+from PySide2.QtWidgets import QListWidget
+from pytestqt import qtbot
+import pytest
+from nexus_constructor.component_fields import FieldWidget
+from nexus_constructor.stream_fields_widget import StreamFieldsWidget
+from tests.ui_tests.test_ui_add_component_window import (
+    add_component_dialog,
+    template,
+    mock_pixel_options,
+)
+
+
+@pytest.fixture
+def create_stream_widget(add_component_dialog):
+    class DummyField:
+        @property
+        def name(self):
+            return "test"
+
+    field = add_component_dialog.create_new_ui_field(DummyField())
+    widget = StreamFieldsWidget(field.attrs_dialog)
+    return widget
+
+
+def test_ui_stream_field_GIVEN_f142_is_selected_WHEN_combo_is_changed_THEN_value_units_edit_is_shown(
+    qtbot
+):
+    listwidget = QListWidget()
+    field = FieldWidget(["test"], listwidget)
+    field_name = "test"
+    field.name = field_name
+
+    widget = StreamFieldsWidget(field)
+
+    qtbot.addWidget(widget)
+
+    widget.schema_combo.currentTextChanged.emit("f142")
+
+    assert widget.value_units_edit.isEnabled()
+    assert widget.value_units_label.isEnabled()
+
+
+def test_ui_stream_field_GIVEN_value_units_is_specified_WHEN_getting_stream_group_from_widget_THEN_value_units_appears_as_dataset(
```

Comment:
"as_dataset" in the test name should be "as_field" to avoid mixing JSON and HDF5 terminology